solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1361,8 +1369,13 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         Fraction memory _loss,
         Fraction memory _level
     ) internal pure returns (uint256) {
-        (uint256 lossN, uint256 lossD) = Math.reducedRatio(_loss.n, _loss.d, MAX_UINT128);
-        return _total.mul(lossD.sub(lossN)).div(lossD).add(_amount.mul(lossN.mul(_level.n)).div(lossD.mul(_level.d)));
+        uint256 levelN = _level.n.mul(_amount);
+        uint256 levelD = _level.d;
+        uint256 max = _total;
+        if (max < levelN) max = levelN;
```

Comment:
We really want to avoid this style. It's harder to read and easy to miss out

Reply:
Yeah, we technically need a `max` function in this file (useful 2 times in this function, and 1 more time in another function).

Reply:
But I wanted to minimize the number of visible changes in this PR, in order to make it clearer.
Then, adding a `max` function separately would be a pretty cosmetic change by itself.

Reply:
I meant that same-line bracketless `if` 😅

Reply:
Yeah, me too. Those two lines should be reformatted as `uint256 maxVal = max(max(levelN, levelD), _total);`.

Reply:
Fixed; please review.